cmd/minishift/cmd/start.go

```diff
@@ -231,18 +235,76 @@ func postClusterUp(machineName string, ip string, port int, routingSuffix string
 		atexit.Exit(1)
 	}
 
-	applyAddOns(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander)
+	addOnManager := addon.GetAddOnManager()
+	configurePersistentVolumes(addOnManager, sshCommander, ocRunner)
+	applyAddOns(addOnManager, ip, routingSuffix, ocPath, kubeConfigPath, sshCommander)
 }
 
-func applyAddOns(ip string, routingSuffix string, ocPath string, kubeConfigPath string, sshCommander provision.SSHCommander) {
-	addOnManager := addon.GetAddOnManager()
+func applyAddOns(addOnManager *manager.AddOnManager, ip string, routingSuffix string, ocPath string, kubeConfigPath string, sshCommander provision.SSHCommander) {
 	err := addOnManager.Apply(addon.GetExecutionContext(ip, routingSuffix, ocPath, kubeConfigPath, sshCommander))
 	if err != nil {
 		fmt.Println("Error executing addon commands: ", err)
 		atexit.Exit(1)
 	}
 }
 
+// TODO - persistent volume creation should really be fixed upstream, aka 'cluster up'. See https://github.com/openshift/origin/issues/14076 (HF)
+// configurePersistentVolumes makes sure that the default persistent volumes created by 'cluster up' have the right permissions - see https://github.com/minishift/minishift/issues/856
+func configurePersistentVolumes(addOnManager *manager.AddOnManager, sshCommander provision.SSHCommander, ocRunner *oc.OcRunner) error {
+	// don't apply this if anyuid is not enabled
+	anyuid := addOnManager.Get("anyuid")
+	if anyuid == nil || !anyuid.IsEnabled() {
+		return nil
+	}
+
+	fmt.Print("-- Waiting for persistent volumes to be created ... ")
+
+	hostPvDir := viper.GetString(hostPvDir)
+
+	var out, err *bytes.Buffer
+
+	// pull the status of the persistent-volume-setup job to determine when the persitent volume creates is completed
+	for {
+		out = new(bytes.Buffer)
+		err = new(bytes.Buffer)
+		exitStatus := ocRunner.Run("get job persistent-volume-setup -n default -o 'jsonpath={ .status.active }'", out, err)
+
+		if exitStatus != 0 || len(err.String()) > 0 {
+			return errors.New("Unable to monitor persistent volume creation")
+		}
+
+		if out.String() != "1" {
+			break
+		}
+
+		time.Sleep(1 * time.Second)
+	}
+
+	// verify the job succeeded
+	out = new(bytes.Buffer)
+	err = new(bytes.Buffer)
+	exitStatus := ocRunner.Run("get job persistent-volume-setup -n default -o 'jsonpath={ .status.succeeded }'", out, err)
+
+	if exitStatus != 0 || len(err.String()) > 0 || out.String() != "1" {
+		return errors.New("Persistent volume creation failed")
+	}
+
+	cmd := fmt.Sprintf("sudo chmod -R 777 %s/pv*", hostPvDir)
+	sshCommander.SSHCommand(cmd)
+
+	// if we have SELinux enabled we need to sort things out there as well
+	cmd = fmt.Sprintf("sudo which chcon; if [ $? -eq 0 ]; then chcon -R -t svirt_sandbox_file_t %s/*; fi", hostPvDir)
```

Comment:
@hferentschik @csrwng If we do not need to set SELinux context because `cluster up` does it then we should  remove this as this would mean we are expecting `cluster up` to set the proper context which is the right thing to do IMO. It would also make this function's function much simpler.